### PR TITLE
Feature fixes

### PR DESCRIPTION
**Pull request by Salamandar**
The decorators now handle correctly the kwargs.

I'm not really fond of the dummy function
```
 +        def build_target_decorator_caller(self, node, args, kwargs):
+            return True 
```
But it's required to comply to the `_get_callee_args`. Or else we should duplicate the `@FeatureNewKwargs` between all the build targets.

### CODE REVIEW

File: mesonbuild/interpreter.py [hunk 1]
```diff
@@ -448,6 +448,7 @@ def partial_dependency_method(self, args, kwargs):
         return DependencyHolder(self.held_object.get_partial_dependency(**kwargs))
 
 class GeneratorHolder(InterpreterObject, ObjectHolder):
+    @FeatureNewKwargs('generator', '0.43.0', 'capture')
```
Comment (nirbheek) [inline mesonbuild/interpreter.py]: @jon-turney added a commit in the other PR that makes this an array: https://github.com/mesonbuild/meson/pull/3661/commits/fb7168635dc551139fd2bae20e750cfce9e06416

 Is that not correct?

Comment (Salamandar) [reply mesonbuild/interpreter.py]: True, that is correct, but I removed everything from build.py to interpreter.py. I'll fix that right after.

**Comment by nirbheek** [comment]
Conflicts, please fix!

**Comment by nirbheek** [comment]
> The last commit resolves #3660 .

If you add that to the commit, it will close the issue automatically.

**Comment by Salamandar** [comment]
That's a neat Github feature. Well I guess that's what we want ;)

**Comment by nirbheek** [comment]
Also, please don't merge master into your branch. It's a little more work to resolve conflicts, but it makes it easier for everyone else to understand the history.

File: mesonbuild/build.py [hunk 2]
```diff
@@ -24,8 +24,8 @@
 from .mesonlib import typeslistify, stringlistify, classify_unity_sources
 from .mesonlib import get_filenames_templates_dict, substitute_values
 from .mesonlib import for_windows, for_darwin, for_cygwin, for_android, has_path_sep
-from .compilers import is_object, clink_langs, sort_clink, lang_suffixes
-from .interpreterbase import FeatureNew, FeatureNewKwargs
+from .compilers import is_object, clike_langs, sort_clink, lang_suffixes
```
Comment (ignatenkobrain) [inline mesonbuild/build.py]: __[Flake8]__

[F401] '.compilers.clike_langs' imported but unused

**Comment by Salamandar** [comment]
@ignatenkobrain Sure but that's not my code ;)

**Comment by nirbheek** [comment]
> @ignatenkobrain Sure but that's not my code ;)

That's automated, and yes it was my code ;)

**Comment by nirbheek** [comment]
This should be merged if/when the CI passes.

**Pull request merged**